packages/frontpage/lib/data/db/post.ts

```diff
@@ -256,3 +276,50 @@ export async function unauthed_deletePost({
   });
   console.log("Done deleting post transaction");
 }
+
+type ModeratePostInput = {
+  rkey: string;
+  authorDid: DID;
+  cid: string;
+  hide: boolean;
+};
+export async function moderatePost({
+  rkey,
+  authorDid,
+  cid,
+  hide,
+}: ModeratePostInput) {
+  console.log(`Moderating post, setting hidden to ${hide}`);
+  await db
+    .update(schema.Post)
+    .set({ status: hide ? "moderator_hidden" : "live" })
+    .where(
+      and(
+        eq(schema.Post.rkey, rkey),
+        eq(schema.Post.authorDid, authorDid),
+        eq(schema.Post.cid, cid),
+      ),
+    );
+}
+
+export async function getPostFromComment({
+  rkey,
+  did,
+}: {
+  rkey: string;
+  did: DID;
+}) {
```

Comment:
We should wrap this in `cache()` and change to positional arguments of `rkey: string, did: DID`.